### PR TITLE
Alert on 0x INSUFFICIENT_LIQUIDITY Error, Prevent swap button flickering on swap screen.

### PR DESCRIPTION
**Pull request by 0xDaedalus**
https://user-images.githubusercontent.com/94649004/179776496-7ee7a3e0-189a-4017-b0eb-0221e09371a6.mov

### CODE REVIEW

File: background/redux-slices/0x-swap.ts [hunk 1]
```diff
@@ -14,6 +19,20 @@ import { getProvider } from "./utils/contract-utils"
 import { ERC20_ABI } from "../lib/erc20"
 import { COMMUNITY_MULTISIG_ADDRESS, ETHEREUM, POLYGON } from "../constants"
 import { EVMNetwork } from "../networks"
+import { setSnackbarMessage } from "./ui"
+
+// @TODO Use ajv validators in conjunction with these types
```
Comment (Shadowfiend) [inline background/redux-slices/0x-swap.ts:24]: Why TODO? This shouldn't be particularly time consuming for these simple types right?

Comment (mhluongo) [reply background/redux-slices/0x-swap.ts:24]: I think the idea here is to patch the issue for a release, then follow up with a deeper fix.

File: background/redux-slices/0x-swap.ts [hunk 2]
```diff
@@ -246,6 +265,29 @@ export const fetchSwapQuote = createBackgroundAsyncThunk(
...
+        // @TODO Extend this to handle more errors
+        parsedBody.validationErrors.find(
+          (e) => e.reason === "INSUFFICIENT_ASSET_LIQUIDITY"
+        )
+      ) {
+        dispatch(setSnackbarMessage("Price Impact Too High"))
+      }
+    }
+  } catch (e) {
+    logger.warn("0x Api Response Parsing Failed")
```
Comment (mhluongo) [inline background/redux-slices/0x-swap.ts:287]: Nit, the capitalization here is very odd, and it doesn't include enough info to debug. Maybe prefer "0x API response parsing failed." and include `error`?

Comment (mhluongo) [reply background/redux-slices/0x-swap.ts:287]: Ah, you're doing that up the callstack. Perhaps "0x API error response parsing failed." then

File: ui/pages/Swap.tsx [hunk 3]
```diff
@@ -338,6 +338,22 @@ export default function Swap(): ReactElement {
...
+      ) {
+        setSellAmount("")
+        return
+      }
+
+      if (
+        "sellAmount" in quoteRequest.amount &&
+        Number(quoteRequest.amount.sellAmount) === 0
+      ) {
+        setBuyAmount("")
```
Comment (mhluongo) [inline ui/pages/Swap.tsx:353]: :+1:

**Review by mhluongo — APPROVED** [review]
:+1: for merge + opening a tech debt issue for the `ajv` validation. Hoping for one more :+1: to merge

**Review by greg-nagy — APPROVED** [review]
✨

https://github.com/tallycash/extension/issues/1841 - follow up ticket

**Comment by greg-nagy** [comment]
I tried to create a more complete fix before merging and cutting the release, but let's cut the release so we can QA and see what else can we include tomorrow

**Pull request merged**